src/search_iterator.cpp

```diff
@@ -239,10 +239,16 @@ int search_iterator::get_fileIndex() const {
 }
 
 search_iterator::reference search_iterator::operator*() const {
+    if (! internal ) {
+        throw EntryNotFound("Cannot get a entry for a uninitialized iterator");
```

Comment:
Is the `EntryNotFound` exception appropriate here? Using an uninitialized iterator is definitely a bug.

Reply:
Now we throw a std::runtime_error.